RAID/refresh_secondary

```diff
@@ -67,170 +69,184 @@ if ! mdadm --detail /dev/md0 2>&1 | grep -q ", degraded $" ; then
-	echo "ERROR: \"Secondary\" disk $part2 TIME is ahead of \"Primary\" disk /dev/sda2"
-	echo "(You should use blank_secondary to reset the second disk)"
+if [ -b /dev/"$SDB"2 ] && [ $(mdadm --examine -Y /dev/"$SDA"2 | grep TIME | cut -d'=' -f2) -lt $(mdadm --examine -Y /dev/"$SDB"2 | grep TIME | cut -d'=' -f2) ]; then
+	echo "ERROR: \"Secondary\" disk /dev/"$SDB"2 TIME is ahead of \"Primary\" disk /dev/"$SDA"2"
+	echo "(You may want to use blank_secondary to erase the second disk first)"
 	exit 1
 fi
-if [ -b $part2 ] && [ $(mdadm --examine -Y /dev/sda2 | grep EVENTS | cut -d'=' -f2) -lt $(mdadm --examine -Y $part2 | grep EVENTS | cut -d'=' -f2) ]; then
-	echo "ERROR: \"Secondary\" disk $part2 EVENT count is higher than the \"Primary\" disk /dev/sda2"
-	echo "(You should use blank_secondary to reset the second disk)"
+if [ -b /dev/"$SDB"2 ] && [ $(mdadm --examine -Y /dev/"$SDA"2 | grep EVENTS | cut -d'=' -f2) -lt $(mdadm --examine -Y /dev/"$SDB"2 | grep EVENTS | cut -d'=' -f2) ]; then
+	echo "ERROR: \"Secondary\" disk /dev/"$SDB"2 EVENT count is higher than the \"Primary\" disk /dev/"$SDA"2"
+	echo "(You may want to use blank_secondary to erase the second disk first)"
 	exit 1
 fi
 
 # Check if "Secondary" disk has been booted separately
-if [ -b $part2 ] && [ "$(mdadm --examine $part2 | grep "Array State" | cut -d' ' -f7)" != "AA" ]; then
-	echo "ERROR: \"Secondary\" disk $part2 has been used separately since last in sync"
-	echo "(You should use blank_secondary to reset the second disk)"
+if [ -b /dev/"$SDB"2 ] && [ "$(mdadm --examine /dev/"$SDB"2 | grep "Array State" | cut -d' ' -f7)" != "AA" ]; then
+	echo "ERROR: \"Secondary\" disk /dev/"$SDB"2 has been used separately since last in sync"
+	echo "(You may want to use blank_secondary to erase the second disk first)"
 	exit 1
 fi
 
 # Ensure the Primary disk is mounted on /boot/efi (ie. not Secondary) if applicable
 if [ $boot = "EFI" ]; then
-	umount /boot/efi
-	mount /dev/sda1 /boot/efi
+	[ $DRY_RUN -eq 0 ] && umount /boot/efi
+	[ $DRY_RUN -eq 0 ] && mount /dev/"$SDA"1 /boot/efi
 fi
 
 
 # ------------- Everything looks sane so let's set to work ----------------
 
 # First repartition the Secondary disk by cloning the Primary's table if blank
-if [ "$(partprobe -s /dev/sda | cut -d' ' -f2-)" != "$(partprobe -s $disk | cut -d' ' -f2-)" ]; then
-	if [ "$(partprobe -s $disk | cut -d' ' -f3-)" = "partitions" ]; then
-		echo -n "Partitioning blank \"Secondary\" disk $disk ... "
+if [ "$(partprobe -s /dev/"$SDA" | cut -d' ' -f2-)" != "$(partprobe -s /dev/$SDB | cut -d' ' -f2-)" ]; then
+	if [ "$(partprobe -s /dev/$SDB | cut -d' ' -f3-)" = "partitions" ]; then
+		echo -n "Partitioning blank \"Secondary\" disk /dev/$SDB ... "
 		if [ $boot = "MBR" ]; then
-			sfdisk -d /dev/sda | sfdisk /dev/sdb > /dev/null
+			[ $DRY_RUN -eq 0 ] && sfdisk -d /dev/"$SDA" | sfdisk /dev/sdb > /dev/null
```

Comment:
It feels like a cheap shot, but maybe `sfdisk /dev/"$SDB"`

Reply:
Good catch - I'd missed that one. It could conceivably have messed up the primary by over-writing its partition table, when the disks have come up out of order,